File: backs/fastapi/src/models/team.py
from sqlalchemy import Column, ForeignKey, Integer, String

from src.models.base import Base


class Team(Base):
    __tablename__ = "team"

    id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    name = Column(String, nullable=False)
    owner = Column(String, nullable=False)

    pokemon_1 = Column(Integer, ForeignKey("pokemon.id"), nullable=False)
    pokemon_2 = Column(Integer, ForeignKey("pokemon.id"))
    pokemon_3 = Column(Integer, ForeignKey("pokemon.id"))
    pokemon_4 = Column(Integer, ForeignKey("pokemon.id"))
    pokemon_5 = Column(Integer, ForeignKey("pokemon.id"))
    pokemon_6 = Column(Integer, ForeignKey("pokemon.id"))

File: backs/symfony/src/Entity/Team.php
<?php

namespace App\Entity;

use ApiPlatform\Metadata\ApiResource;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity]
#[ApiResource]
class Team
{
    #[ORM\Id, ORM\Column, ORM\GeneratedValue]
    private ?int $id = null;

    #[ORM\Column]
    #[Assert\NotNull]
    #[Assert\NotBlank]
    public string $name;

    #[ORM\Column]
    #[Assert\NotNull]
    #[Assert\NotBlank]
    public string $owner;

    #[ORM\ManyToOne]
    #[Assert\NotBlank]
    #[Assert\NotNull]
    public Pokemon $pokemon1;

    #[ORM\ManyToOne]
    public ?Pokemon $pokemon2;

    #[ORM\ManyToOne]
    public ?Pokemon $pokemon3;

    #[ORM\ManyToOne]
    public ?Pokemon $pokemon4;

    #[ORM\ManyToOne]
    public ?Pokemon $pokemon5;

    #[ORM\ManyToOne]
    public ?Pokemon $pokemon6;

    public function getId(): ?int
    {
        return $this->id;
    }
}
